app/src/main/java/com/teamagam/gimelgimel/app/notifications/AppNotifier.java

```diff
@@ -53,41 +60,71 @@ public void notifyNewMessage(ChatMessage chatMessage) {
     mAlertText = context.getString(R.string.alert_notification_text);
   }
 
-  private void initializeMessageStrings(Context context) {
+  private void initMessageStrings(Context context) {
     mMessageTitle = context.getString(R.string.message_notification_title);
     mMessageSummary = context.getString(R.string.message_notification_summary);
     mMessageText = context.getString(R.string.message_notification_text);
   }
 
-  private void notifyAlert() {
-    mBuilder.setSmallIcon(R.drawable.ic_alert_notification);
-    mBuilder.setStyle(createStyle(mAlertTitle, mAlertSummary, mAlertText));
-    mBuilder.setContentTitle(mAlertTitle);
-    mBuilder.setContentText(mAlertText);
+  private void initPrefKeys(Context context) {
+    mShouldVibratePrefKey = context.getString(R.string.pref_notifications_vibration_key);
+    mRingtonePrefKey = context.getString(R.string.pref_notifications_ringtone_key);
+  }
 
-    Notification n = mBuilder.build();
-    mNotificationManager.notify(Constants.ALERTS_NOTIFICATION_ID, n);
+  private void notifyAlert() {
+    notify(R.drawable.ic_alert_notification, mAlertTitle, mAlertSummary, mAlertText,
+        Constants.ALERTS_NOTIFICATION_ID);
   }
 
   private void notifyMessage() {
-    mBuilder.setSmallIcon(R.drawable.ic_lambda_logo);
-    mBuilder.setStyle(createStyle(mMessageTitle, mMessageSummary, mMessageText));
-    mBuilder.setContentTitle(mMessageTitle);
-    mBuilder.setContentText(mMessageText);
+    notify(R.drawable.ic_lambda_logo, mMessageTitle, mMessageSummary, mMessageText,
+        Constants.MESSAGES_NOTIFICATION_ID);
+  }
+
+  private void notify(int smallIconResId,
+      String notificationTitle,
+      String notificationSummary,
+      String notificationText,
+      int notificationId) {
+    NotificationCompat.Builder builder =
+        createNotificationBuilder(smallIconResId, notificationTitle, notificationSummary,
+            notificationText);
 
-    Notification n = mBuilder.build();
-    mNotificationManager.notify(Constants.MESSAGES_NOTIFICATION_ID, n);
+    mNotificationManager.notify(notificationId, builder.build());
+  }
+
+  private NotificationCompat.Builder createNotificationBuilder(int smallIconResId,
+      String notificationTitle,
+      String notificationSummary,
+      String notificationText) {
+    NotificationCompat.Builder builder = new NotificationCompat.Builder(mContext).setContentIntent(
+        PendingIntent.getActivity(mContext, 0, mMainActivityIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT))
+        .setPriority(PRIORITY_MAX)
+        .setAutoCancel(true)
+        .setSmallIcon(smallIconResId)
+        .setStyle(createStyle(notificationTitle, notificationSummary, notificationText))
+        .setContentTitle(notificationTitle)
+        .setContentText(notificationText)
+        .setSound(getPreferredRingtoneUri());
+
+    if (shouldVibrate()) {
+      builder.setVibrate(VIBRATION_PATTERN);
+    }
+
+    return builder;
```

Comment:
I liked the extraction process u did here, I think you can make this method return the notification itself and not the builder by using build here.